apps/next/middleware.ts

```diff
@@ -1,18 +1,21 @@
-import { authMiddleware } from "@clerk/nextjs";
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
 
-// See https://clerk.com/docs/references/nextjs/auth-middleware
-// for more information about configuring your Middleware
-export default authMiddleware({
-  // Allow signed out users to access the specified routes:
-  // publicRoutes: ['/anyone-can-visit-this-route'],
-  publicRoutes: [
-    "/",
-    "/register",
-    "/login",
-    "/login-to-cli",
-    "/api/doc",
-    "/__nextjs_original-stack-frame",
-  ],
+// For more info, see:
+// https://clerk.com/docs/upgrade-guides/core-2/nextjs#migrating-to-clerk-middleware
```

Comment:
did you also do the `npx @clerk/upgrade` command recommended in this doc?

Reply:
I just did. I also had to rename Metal's own authenticateRequest to authenticateUser because `bunx @clerk/upgrade` was miscontruing it to be instances of Clerk's function with the same name.

You can find all the details here:
https://github.com/onmetal-dev/metal/pull/56/commits/0c75f02ecc0631bf5aa697e62acb90272fc14251

Reply:
And oh I ran `bunx @clerk/upgrade`, not `npx @clerk/upgrade`.